include/nbl/builtin/hlsl/bxdf/base/cook_torrance_base.hlsl

```diff
@@ -378,7 +362,13 @@ struct SCookTorrance
 
         spectral_type quo;
         NBL_IF_CONSTEXPR(IsBSDF)
-            quo = hlsl::promote<spectral_type>(G2_over_G1);
+        {
+            spectral_type prefixThroughputWeights = interaction.getPrefixThroughputWeights();
+            spectral_type reflectance = impl::__implicit_promote<spectral_type, typename fresnel_type::vector_type>::__call(_f(hlsl::abs(cache.getVdotH())));
+            const scalar_type scaled_reflectance = hlsl::dot(reflectance, prefixThroughputWeights);
+            quo = hlsl::mix(reflectance / scaled_reflectance,
+                    (hlsl::promote<spectral_type>(1.0) - reflectance) / (scalar_type(1.0) - scaled_reflectance), cache.isTransmission()) * G2_over_G1;
```

Comment:
btw fun fact, you can do `fresnelVal = mix(reflectance,hlsl::promote<fresnel_type::vector_type>(1.0)-reflectance,cache.isTransmission())` first, and then just compute `fresnelVal/dot(lumaCoeffs,fresnelVal)`

because of the way everything is a linear combo and weights add up to 1

P.S. Obivously don't do the division and dot product if Fresnel is monochrome https://github.com/Devsh-Graphics-Programming/Nabla/pull/918/files/76ad0ed10fb39d65fee94621e339b3b58b5a8f29..946b05082c4f5e13fbaeb112ed974b52d6879f36?diff=unified&w=1#r2498628665